iam-login-service/src/main/webapp/resources/iam/apps/registration/registration.controller.js

```diff
@@ -197,10 +266,53 @@ function RegistrationController(
   }
 
   function fieldReadonly(name) {
-    if ($scope.config) {
-      var field = $scope.config.fields[name];
-      return field.readOnly === true;
+    return $scope.config?.fields?.[name]?.readOnly === true;
+  }
+
+  function populateFieldsWithAdminPreference() {
+    if ($scope.config.fields) {
+      for (let field in $scope.config.fields) {
+        if (
+          $scope.config.fields[field]["fieldBehaviour"].toLowerCase() == "mandatory"
+        ) {
+          $scope.fields[field].showField = true;
+          $scope.fields[field].required = true;
+        } else if (
+          $scope.config.fields[field]["fieldBehaviour"].toLowerCase() === "optional"
+        ) {
+          $scope.fields[field].showField = true;
+          $scope.fields[field].required = false;
+        } else {
+          $scope.fields[field].showField = false;
+          $scope.fields[field].required = false;
+        }
+      }
     }
-    return false;
+  }
+
+  function getFieldErrorMessage(fieldName) {
+    let field = $scope.registrationForm[fieldName];
+    let fieldInfo = $scope.fields[fieldName];
+
+    if (!fieldInfo) {
+        return null;
+    }
+
+    if (field.$error.required) {
+        if (fieldName === 'notes') {
+            return 'Please provide a reason for your registration request';
+        }
+        return 'Please provide ' + fieldInfo.articleToUse + ' ' + fieldInfo.label.toLowerCase();
+    } else if (field.$error.minlength) {
+        return fieldInfo.label + ' must be at least ' + fieldInfo.minlength + ' characters long';
+    } else if (field.$error.email) {
+        return 'This is not a valid email';
+    } else if (field.$error.emailAvailable) {
```

Comment:
I think these two validators (for email and username) are not working correctly. I am able to enter an existing username or email without getting any errors. Only after clicking on ‘Register’ do I get an empty red box at the bottom of the page and the exception is raised saying that the email or username already exists.